File: java/robot-simulator/_iterations/1/src/main/java/Orientation.java
enum Orientation {
    NORTH, EAST, SOUTH, WEST;

    Orientation right() {
        if (this == NORTH) {
            return EAST;
        } else if (this == EAST) {
            return SOUTH;
        } else if (this == SOUTH) {
            return WEST;
        } else {
            return NORTH;
        }
    }

    Orientation left() {
        if (this == NORTH) {
            return WEST;
        } else if (this == WEST) {
            return SOUTH;
        } else if (this == SOUTH) {
            return EAST;
        } else {
            return NORTH;
        }
    }
}

File: kotlin/robot-simulator/src/main/kotlin/Orientation.kt
enum class Orientation {
    NORTH, EAST, SOUTH, WEST;

    val left: Orientation get() = when (this) {
        NORTH -> WEST
        WEST -> SOUTH
        SOUTH -> EAST
        EAST -> NORTH
    }

    val right: Orientation get() = when (this) {
        NORTH -> EAST
        EAST -> SOUTH
        SOUTH -> WEST
        WEST -> NORTH
    }
}
